Client/Frontend/Browser/FrequencyQuery.swift

```diff
@@ -7,28 +7,34 @@ import Foundation
 import Shared
 import Storage
 import Data
+import BraveCore
 
 class FrequencyQuery {
     
-    private static let queue = DispatchQueue(label: "frequency-query-queue")
-    private static var cancellable: DispatchWorkItem?
+    private let historyAPI: BraveHistoryAPI
+    private let bookmarkAPI: BraveBookmarksAPI
     
-    public static func sitesByFrequency(containing query: String? = nil,
-                                        completion: @escaping (Set<Site>) -> Void) {
-        
-        Historyv2.byFrequency(query: query) { historyList in
+    private let queue = DispatchQueue(label: "frequency-query-queue")
+    private var cancellable: DispatchWorkItem?
+    
+    init(historyAPI: BraveHistoryAPI, bookmarkAPI: BraveBookmarksAPI) {
+        self.historyAPI = historyAPI
+        self.bookmarkAPI = bookmarkAPI
+    }
+    
+    public func sitesByFrequency(containing query: String? = nil, completion: @escaping (Set<Site>) -> Void) {
+        historyAPI.byFrequency(query: query) { [weak self] historyList in
             let historySites = historyList
-                .map { Site(url: $0.url ?? "", title: $0.title ?? "") }
-            
-            cancellable = DispatchWorkItem {
+                .map { Site(url: $0.url.absoluteString ?? "", title: $0.title ?? "") }
+
+            self?.cancellable = DispatchWorkItem {
```

Comment:
Probably should guard let self at the beginning so you don't start work with self dead

Reply:
Fix at https://github.com/brave/brave-ios/pull/4024/commits/7bd541f3705f05fe5e634ff51adf1e05fb928695